tests/conftest.py

```diff
@@ -38,27 +42,91 @@
 
 
 @pytest.fixture(scope="session")
-def _raw_client(request: pytest.FixtureRequest) -> Client:
-    path = os.environ.get("TREZOR_PATH")
-    interact = int(os.environ.get("INTERACT", 0))
-    if path:
-        try:
-            transport = get_transport(path)
-            return Client(transport, auto_interact=not interact)
-        except Exception as e:
-            request.session.shouldstop = "Failed to communicate with Trezor"
-            raise RuntimeError(f"Failed to open debuglink for {path}") from e
+def emulator(request: pytest.FixtureRequest) -> Generator[Emulator, None, None]:
+    """Fixture for getting emulator connection in case tests should operate it on their own.
+
+    Is responsible for starting it at the start of the session and stopping
+    it at the end of the session - using `with EmulatorWrapper...`.
+
+    Makes sure that each process will run the emulator on a different
+    port and with different profile directory, which is cleaned afterwards.
+
+    Used so that we can run the device tests in parallel using `pytest-xdist` plugin.
+    Docs: https://pypi.org/project/pytest-xdist/
+
+    NOTE for parallel tests:
+    So that all worker processes will explore the tests in the exact same order,
+    we cannot use the "built-in" random order, we need to specify our own,
+    so that all the processes share the same order.
+    Done by appending `--random-order-seed=$RANDOM` as a `pytest` argument,
+    using system RNG.
+    """
+
+    model = str(request.session.config.getoption("model"))
+    interact = os.environ.get("INTERACT") == "1"
+
+    assert model in ("core", "legacy")
+    if model == "legacy":
+        raise RuntimeError(
+            "Legacy emulator is not supported until it can be run on arbitrary ports."
+        )
+
+    def _get_port() -> int:
+        """Get a unique port for this worker process on which it can run.
+
+        Guarantees to be unique because each worker has a different name.
+        gw0=>20000, gw1=>20111, gw2=>20222, etc.
+
+        Assumes there will not be more than 10 workers.
```

Comment:
me and my 16 cores don't like this ;) 

how about instead:
```
assert worker_id.startswith("gw")
return 20000 + int(worker_id[2:]) * 3
```

Reply:
to expand: `*3` numerically because the emulator eats 3 consecutive ports: Normal link, Debug link, Webauthn fake interface.

Reply:
Works for me. Done in [55c719b](https://github.com/trezor/trezor-firmware/pull/2462/commits/55c719b512eaa92b53a42975378af244bcf68955)